lib/validator.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const Joi = require('joi');
+
+module.exports = {
+  auth: Joi.object().keys({
+    credentials: Joi.object().required()
+  }),
```

Comment:
But if hapi's authentication strategy failed then I don't think it would actually ever get to execute this plugin, right? I think by having validation against `auth.credentials` then we are really just asserting that hapi is doing the right thing, which we shouldn't need to do. 

Surely for our use case this validation is fine, but I was thinking more about if this was used elsewhere in a project that didn't care about authentication. Would the plugin just fail for them, even if they don't actually care about the `auth` object at all?